crates/typst-eval/src/flow.rs

```diff
@@ -17,8 +17,8 @@ pub enum FlowEvent {
     /// Skip the remainder of the current iteration in a loop.
     Continue(Span),
     /// Stop execution of a function early, optionally returning an explicit
-    /// value.
-    Return(Span, Option<Value>),
+    /// value. The final boolean indicates whether the return was conditional.
+    Return(Span, Option<Value>, bool),
```

Comment:
This way to detect conditionalness does seem sensible in principle. I'll need to experiment a bit to see whether it could result in false positive. I'll do that after my vacation.